lib/segment/src/index/hnsw_index/graph_layers_builder.rs

```diff
@@ -509,28 +497,15 @@ impl GraphLayersBuilder {
         &self,
         point_id: PointOffsetType,
         curr_level: usize,
-        visited_list: &VisitedListHandle,
         points_scorer: &FilteredScorer,
-        mut search_context: SearchContext,
+        nearest: FixedLengthPriorityQueue<ScoredPointOffset>,
     ) {
         let level_m = self.hnsw_m.level_m(curr_level);
         let scorer = |a, b| points_scorer.score_internal(a, b);
 
         let selected_nearest = {
+            let iter = nearest.into_iter_sorted();
             let mut existing_links = self.links_layers[point_id as usize][curr_level].write();
-            {
-                let ready_list = self.ready_list.read();
-                for existing_link in existing_links.iter() {
-                    if !visited_list.check(existing_link) && ready_list[existing_link as usize] {
```

Comment:
Great that you proved that this code is unreachable. We also need to remove this logic from gpu implementation in the future